### PR TITLE
build: Fix build failure with CI in FreeBSD

### PR DESCRIPTION
**Pull request by erw7**
Ref. https://builds.sr.ht/~jmk/job/257200#task-build-624

### CODE REVIEW

**Comment by tjdevries** [comment]
Do you know why it's not finding the usage in the assert? It's very weird to me that it doesn't find it.

**Comment by tjdevries** [comment]
It gets used about 10 lines below there.

**Comment by jamessan** [comment]
> Do you know why it's not finding the usage in the assert?

`assert()` gets compiled out in release builds.

File: src/nvim/lua/executor.c [hunk 1]
```diff
@@ -1406,6 +1406,7 @@ char_u *nlua_register_table_as_callable(typval_T *const arg)
   lua_State *const lstate = nlua_enter();
 
   int top = lua_gettop(lstate);
```
Comment (jamessan) [inline src/nvim/lua/executor.c:1408]: What about changing this to

```suggestion
#ifndef NDEBUG
  int top = lua_gettop(lstate);
#endif
```

File: src/nvim/lua/executor.c [hunk 2]
```diff
@@ -1406,6 +1406,7 @@ char_u *nlua_register_table_as_callable(typval_T *const arg)
...
+  (void)top;
```
Comment (jamessan) [inline src/nvim/lua/executor.c:1409]: What about this?

```suggestion
#ifndef NDEBUG
  int top = lua_gettop(lstate);
#endif
```

Comment (tjdevries) [reply src/nvim/lua/executor.c:1409]: Ahhhh, I see now. Very cool. Thanks, that's what I was looking for!

**Pull request merged**